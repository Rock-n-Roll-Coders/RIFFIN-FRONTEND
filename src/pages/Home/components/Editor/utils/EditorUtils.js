import { STEP_COUNT, MIN_BLOCK_COLS, MAX_BLOCK_COLS } from '../EditorConfig'

const calcNewMaxLength = (numberOfCols, numberOfStrings) => (numberOfCols * numberOfStrings) + (numberOfStrings * 2) - 1
const calculateNewFirstCol = (numberOfCols, index) => (numberOfCols + 1) * (index)
const calculateNewLastCol = (numberOfCols, stringNum, index) => (numberOfCols * stringNum) + index

/* 
  Needs an actions object with the following properties:
  area
  characterToAdd
  action
  cols
*/
export const updateBlockValue = (action) => {
  if((action.action === "increase" && action.cols === MAX_BLOCK_COLS) ||
    (action.action === "decrease" && action.cols === MIN_BLOCK_COLS)) {
    return action.area
  }

  const guitarStrings = action.area.split('\n')
  let newValueString = ""
  for(let i = 0; i < guitarStrings.length; i++) {
    const array = [...guitarStrings[i]]
    for(let j = 0; j < STEP_COUNT; j++) {
      if(action.action === "increase") {
        array.push(action.characterToAdd)
      } else {
        array.pop()
      }
    }
    if(i !== guitarStrings.length - 1) {
      array.push("\n")
    }
    newValueString += array.join("")
  }
  return newValueString
}

/* 
  action {
    cols,
    maxLength,
    action,
    stringCount
  }
*/
export const updateTextAreaAttributes = (action) => {
  if((action.action === "increase" && action.cols === MAX_BLOCK_COLS) ||
    (action.action === "decrease" && action.cols === MIN_BLOCK_COLS)) {
    return {
      cols: action.cols,
      maxLength: action.maxLength
    }
  }
  if(action.action === "increase") {
    return {
      cols: action.cols + STEP_COUNT,
      maxLength: calcNewMaxLength(action.cols + STEP_COUNT, action.stringCount)
    }
  } else {   
    return {
      cols: action.cols - STEP_COUNT,
      maxLength: calcNewMaxLength(action.cols - STEP_COUNT, action.stringCount)
    }
  }
}


export const getMapOfLastColumnIndexes = (action) => {
  const newLastCols = {}
  for(let i = 0; i < action.stringCount; i++) {
    const stringNum = i + 1
    const newVal = calculateNewLastCol(action.cols, stringNum, i)
    newLastCols[newVal] = true
  }
  return newLastCols
}

export const getMapOfFirstColumnIndexes = (action) => {
  const newFirstCols = {}
  for(let i = 0; i < action.stringCount; i++) {
    const newVal = calculateNewFirstCol(action.cols, i)
    newFirstCols[newVal] = true
  }
  return newFirstCols
}