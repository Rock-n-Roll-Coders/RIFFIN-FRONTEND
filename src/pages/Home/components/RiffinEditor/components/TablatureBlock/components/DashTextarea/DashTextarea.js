// Components / hooks
import { useTheme } from "@mui/material/styles";

/**
 * * DashTextarea is readonly and displays the dashes behind the input textarea. It is absolutely positioned and uses a z-index to ensure the user can't interact with the textarea (the textarea is readonly for an additional safeguard). The values are derived from the user actions in the InputTextarea.
 */

const DashTextarea = (props) => {
  const theme = useTheme();
  const dashesStyle = {
    background: "transparent",
    margin: 0,
    position: "absolute",
    resize: "none",
    top: "0",
    left: "0",
    translate: "0",
    zIndex: 1,
    outline: "none",
    border: "none",
    fontFamily: "Fira Code",
    color: theme.palette.dashes.main,
    fontSize: "inherit",
    padding: 0,
  };

  return (
    <textarea
      readOnly={true}
      style={dashesStyle}
      value={props.block.dashes}
      cols={props.block.cols}
      rows={props.numberOfStrings}
      maxLength={props.block.maxLength}
    />
  );
};
 
export default DashTextarea;