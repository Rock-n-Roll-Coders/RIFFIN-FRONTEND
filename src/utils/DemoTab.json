{
    "_id": "demo",
    "owner": "",
    "numberOfStrings": 6,
    "name": "Riffin Blues (demo tab)",
    "blocks": [
        {
            "label": "Riffin Editor core features:\n- Change the size of the staff using the Staff Menu ===>\n\n- Take advantage of the Duplicate Chord feature by pressing \"]\". This will duplicate an entire chord starting at the position directly behind the cursor.\n\n- Rearrange staffs by dragging the handle (=) on the left edge.\n\n- Above each staff is an area for notes!\n\n- Catalog your tablature with Tags. Tags can be added to the bar at the top of the screen.\n\nSAMPLE TAB: \nTuning: EADGBE\n\n\n\nGuitar 1:\nRepeat twice for 8 bars - add some bluesy flare to that strummin!",
            "inputs": "|                                      |\n|                                      |\n|                                      |\n|                                      |\n| 2 2 4 4  2 2 4 4  2 2 4 4  2 2 4 4   |\n| 0 0 0 0  0 0 0 0  0 0 0 0  0 0 0 0   |",
            "dashes": "|--------------------------------------|\n|--------------------------------------|\n|--------------------------------------|\n|--------------------------------------|\n|- - - - -- - - - -- - - - -- - - - ---|\n|- - - - -- - - - -- - - - -- - - - ---|",
            "cols": 40,
            "maxLength": 251,
            "_id": "637bdb2aa557ebd717828c1e"
        },
        {
            "label": "Guitar 1:\nNow throw in 4 bars of that sweet open A power chord.\nGotta throw in some bluesy rhythm!",
            "inputs": "|                                      |\n|                                      |\n|                                      |\n| 2 2 4 4  2 2 4 4  2 2 4 4 5 5 4 4    |\n| 0 0 0 0  0 0 0 0  0 0 0 0 0 0 0 0    |\n|                                      |",
            "dashes": "|--------------------------------------|\n|--------------------------------------|\n|--------------------------------------|\n|- - - - -- - - - -- - - - - - - - ----|\n|- - - - -- - - - -- - - - - - - - ----|\n|--------------------------------------|",
            "cols": 40,
            "maxLength": 251,
            "_id": "637be4bd6065d9183d028a91"
        },
        {
            "label": "Guitar 1:\nBack to the start again... ",
            "inputs": "|                                      |\n|                                      |\n|                                      |\n|                                      |\n| 2 2 4 4  2 2 4 4  2 2 4 4  2 2 4 4   |\n| 0 0 0 0  0 0 0 0  0 0 0 0  0 0 0 0   |",
            "dashes": "|--------------------------------------|\n|--------------------------------------|\n|--------------------------------------|\n|--------------------------------------|\n|- - - - -- - - - -- - - - -- - - - ---|\n|- - - - -- - - - -- - - - -- - - - ---|",
            "cols": 40,
            "maxLength": 251,
            "_id": "637bdb2aa557ebd717828c1e"
        },
        {
            "label": "Guitar 1:\nWrap up with a bluesy walk-back!",
            "inputs": "|                                                          1~~  |\n|                                                          0~~  |\n|                                                          2~~  |\n|  4 4 4 4 4 4 2~~ 2 2 2 2 2                               1~~  |\n|  2 2 2 2 2 2 0~~ 0 0 0 0 0 2 2 5 5 5 4 4 4 3 3 3 2 0h1h2 2~~  |\n|                            0 0 0 0 0 0 0 0 0 0 0 0            |",
            "dashes": " ----------------------------------------------------------   --|\n|----------------------------------------------------------   --|\n|----------------------------------------------------------   --|\n|-- - - - - - -   - - - - - -------------------------------   --|\n|-- - - - - - -   - - - - - - - - - - - - - - - - - -     -   --|\n|---------------------------- - - - - - - - - - - - ------------|",
            "cols": 65,
            "maxLength": 401,
            "_id": "637be4bd6065d9183d028a93"
        },
        {
            "label": "Guitar 2:\nCome in swinging with some blues licks!",
            "inputs": "|                                                     |\n|   14 / 17 17 17 17 17 17 17 15 17 15                |\n|                                      14/ 16         |\n|                                             14  14~~|\n|                                                     |\n|                                                     |",
            "dashes": "|-----------------------------------------------------|\n|---  - -  -  -  -  -  -  -  -  -  -  ----------------|\n|--------------------------------------   -  ---------|\n|---------------------------------------------  --    |\n|-----------------------------------------------------|\n|-----------------------------------------------------|",
            "cols": 55,
            "maxLength": 341,
            "_id": "637be4bd6065d9183d028a94"
        }
    ],
    "tags": [
        "Blues",
        "ðŸŽ¸",
        "Need to practice"
    ]
}