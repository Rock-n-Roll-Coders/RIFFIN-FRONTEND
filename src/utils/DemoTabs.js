export const demoTabs = [
    {
        "owner": "",
        "numberOfStrings": 6,
        "name": "12 bar blues",
        "blocks": [
            {
                "label": "TUNING: EADGBE\n\nBluesy intro walk",
                "inputs": "|                          1 1 0  |\n|                          0 0 0  |\n|                          2 2    |\n|                          1 1    |\n| 7 7 5 5 4 4 3 3 2 0 1 2  2 2    |\n|                                 |",
                "dashes": "|-------------------------- - - --|\n|-------------------------- - - --|\n|-------------------------- - ----|\n|-------------------------- - ----|\n|- - - - - - - - - - - - -- - ----|\n|---------------------------------|",
                "cols": 35,
                "maxLength": 221,
                "_id": "6382930bff50a5d590725d89",
                "createdAt": "2022-11-26T22:28:27.447Z",
                "updatedAt": "2022-11-26T22:28:27.447Z"
            },
            {
                "label": " ",
                "inputs": "|                                                                              |\n|                                                                              |\n|                                                                              |\n|                     2 2 4 4  2 2 4 4                                         |\n| 2 2 4 4   2 2 4 4   0 0 0 0  0 0 0 0   2 2 4 4  2 2 4 4  2 2 4 4  2 2 4 4    |\n| 0 0 0 0   0 0 0 0                      0 0 0 0  0 0 0 0  0 0 0 0  0 0 0 0    |",
                "dashes": "|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|\n|--------------------- - - - -- - - - -----------------------------------------|\n|- - - - --- - - - --- - - - -- - - - --- - - - -- - - - -- - - - -- - - - ----|\n|- - - - --- - - - ---------------------- - - - -- - - - -- - - - -- - - - ----|",
                "cols": 80,
                "maxLength": 491,
                "_id": "6382930bff50a5d590725d8a",
                "createdAt": "2022-11-26T22:28:27.448Z",
                "updatedAt": "2022-11-26T22:28:27.448Z"
            },
            {
                "label": "  ",
                "inputs": "|                                                                              |\n|                                                                              |\n|                                                                              |\n| 2 2 4 4  2 2 4 4  2 2 4 4  2 2 4 4                                           |\n| 0 0 0 0  0 0 0 0  0 0 0 0  0 0 0 0  2 2 4 4  2 2 4 4  2 2 4 4   2 2 4 4      |\n|                                     0 0 0 0  0 0 0 0  0 0 0 0   0 0 0 0      |",
                "dashes": "|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|\n|- - - - -- - - - -- - - - -- - - - -------------------------------------------|\n|- - - - -- - - - -- - - - -- - - - -- - - - -- - - - -- - - - --- - - - ------|\n|------------------------------------- - - - -- - - - -- - - - --- - - - ------|",
                "cols": 80,
                "maxLength": 491,
                "_id": "6382930bff50a5d590725d8b",
                "createdAt": "2022-11-26T22:28:27.448Z",
                "updatedAt": "2022-11-26T22:28:27.448Z"
            },
            {
                "label": " ",
                "inputs": "|                                        |                                     |\n|                                        |   then repeat from intro walk       |\n|                                        |                                     |\n| 4 4 6 6  4 4 6 6  2 2 4 4  2 2 4 2     |                                     |\n| 2 2 2 2  2 2 2 2  0 0 0 0  0 0 0 0 2   |                                     |\n|                                    0 0 |                                     |",
                "dashes": "|---------------------------------------- -------------------------------------|\n|---------------------------------------- ---    -      -    -     -    -------|\n|---------------------------------------- -------------------------------------|\n|- - - - -- - - - -- - - - -- - - - ----- -------------------------------------|\n|- - - - -- - - - -- - - - -- - - - - --- -------------------------------------|\n|------------------------------------ - - -------------------------------------|",
                "cols": 80,
                "maxLength": 491,
                "_id": "6382930bff50a5d590725d8c",
                "createdAt": "2022-11-26T22:28:27.448Z",
                "updatedAt": "2022-11-26T22:28:27.448Z"
            }
        ],
        "tags": [
            "blues",
            "practice"
        ],
        "_id": "demo_blues_riff",
        "createdAt": "2022-11-26T22:28:27.448Z",
        "updatedAt": "2022-11-26T22:28:27.448Z",
        "__v": 0
    },
    {
      "_id": "riffin_sketchpad_tutorial",
      "owner": "c52209cb-6e0c-49f9-a4b0-cb5dbd25ae58",
      "numberOfStrings": 6,
      "name": "Riffin sketchpad tutorial",
      "blocks": [
          {
              "label": "Click the 'EDIT' button to open this tab in the sketchpad. The sketchpad is only available on large displays, if you're viewing this on mobile you will not see an 'EDIT' button.\n===\n",
              "inputs": "|    0            0          |\n|    0          0   0        |\n|    1        1       1      |\n|    2      2           2    |\n|    2    2               2  |\n|    0  0                   0|",
              "dashes": "|---- ------------ ----------|\n|---- ---------- --- --------|\n|---- -------- ------- ------|\n|---- ------ ----------- ----|\n|---- ---- --------------- --|\n|---- -- ------------------- |",
              "cols": 30,
              "maxLength": 191,
              "_id": "637c3c40c682ca4f60eb9341"
          },
          {
              "tempKey": "Wed Nov 23 2022 20:10:46 GMT-0500 (Eastern Standard Time)0.6697429678666145",
              "inputs": "|                                                |\n|                                                |\n|                                                |\n|                                                |\n|                                                |\n|                                                |",
              "dashes": "|------------------------------------------------|\n|------------------------------------------------|\n|------------------------------------------------|\n|------------------------------------------------|\n|------------------------------------------------|\n|------------------------------------------------|",
              "cols": 50,
              "maxLength": 311,
              "label": "Below is a staff for tablature. Give yourself some space to type - use the \"Staff Menu\" on the right to increase the length, then click anywhere on the staff to start writing! \n\nThe arrow keys are helpful for navigating."
          },
          {
              "label": "Above each staff is an area for notes. Take as many lines as you need.",
              "inputs": "|                                                15/17 17^ (bend all the way!) |\n|                                                                              |\n|   You can write notes on the staff too!                                      |\n|                                                                              |\n|                                                                              |\n|                                                                              |",
              "dashes": "|------------------------------------------------     -   -     -   -   -     -|\n|------------------------------------------------------------------------------|\n|---   -   -     -     -  -   -     -    --------------------------------------|\n|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|\n|------------------------------------------------------------------------------|",
              "cols": 80,
              "maxLength": 491,
              "_id": "637c54d29e1c0a4310a8f3bc"
          },
          {
              "label": "Duplicate a chord behind the cursor by pressing \"]\". Try duplicating the D chord below!\nCheck out the 'Special inputs' panel for more cool tricks.\n",
              "inputs": "| 0| <--chord behind cursor:  0 0           D chord ->  2                      |\n| 0                           0 0                       4                      |\n| 1                           1 1                       2                      |\n| 2                           2 2                       0                      |\n| 2                           2 2                       x                      |\n| 0                           0 0                       x                      |",
              "dashes": "|-  -        -      -       -- - ----------- -     -  -- ----------------------|\n|- --------------------------- - ----------------------- ----------------------|\n|- --------------------------- - ----------------------- ----------------------|\n|- --------------------------- - ----------------------- ----------------------|\n|- --------------------------- - ----------------------- ----------------------|\n|- --------------------------- - ----------------------- ----------------------|",
              "cols": 80,
              "maxLength": 491,
              "_id": "637c54d29e1c0a4310a8f3bd"
          },
          {
              "label": "Use the 'add tag' input at the top of the screen to help organize your tablature.",
              "inputs": "|                                                |\n|                                                |\n<-- use the drag handle (=) to rearrange staffs! |\n|                                                |\n|                                                |\n|                                                |",
              "dashes": "|------------------------------------------------|\n|------------------------------------------------|\n   -   -   -    -      -   -  -         -       -|\n ------------------------------------------------|\n|------------------------------------------------|\n|------------------------------------------------|",
              "cols": 50,
              "maxLength": 311,
              "_id": "637c58127707340387e4d856"
          }
      ],
      "tags": [
          "Practice"
      ]
  }
]