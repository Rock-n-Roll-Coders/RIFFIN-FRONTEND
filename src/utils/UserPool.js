import { CognitoUserPool } from "amazon-cognito-identity-js";
// UserPoolId comes from the general tab in Cognito
// ClientID comes from the App Client Settings in Cognito
const poolData = {
  UserPoolId: process.env.REACT_APP_COGNITO_USER_POOL_ID,
  ClientId: process.env.REACT_APP_COGNITO_CLIENT_ID,
};

export default new CognitoUserPool(poolData);
