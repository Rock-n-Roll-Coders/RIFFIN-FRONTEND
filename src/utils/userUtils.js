const getIdTokenFromUser = (user) => {
  let dataKey = user.userDataKey;
  dataKey = dataKey.replace(".userData", ".idToken");
  const idToken = user.storage[dataKey];
  return idToken;
};

export { getIdTokenFromUser };
