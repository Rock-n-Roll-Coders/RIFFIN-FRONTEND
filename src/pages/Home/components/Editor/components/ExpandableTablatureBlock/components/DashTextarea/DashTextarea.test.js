import { getNewGuitarBlock } from "../../../../utils/EditorUtils";
import DashTextarea from "./DashTextarea";
import renderer from 'react-test-renderer';

// Note that style props may be undefined undefined because I'm not wrapping this in the theme context provider.
test('renders correctly using a new tablatureBlock', ()=> {
  const block = getNewGuitarBlock();
  const tree = renderer
    .create(<DashTextarea block={block}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});