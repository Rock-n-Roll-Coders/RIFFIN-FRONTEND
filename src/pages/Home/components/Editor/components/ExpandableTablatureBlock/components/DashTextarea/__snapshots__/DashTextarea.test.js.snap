// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly using a new tablatureBlock 1`] = `
<textarea
  cols={40}
  maxLength={251}
  readOnly={true}
  style={
    Object {
      "background": "transparent",
      "border": "none",
      "color": undefined,
      "fontFamily": "Fira Code",
      "fontSize": "1.2rem",
      "left": "0",
      "margin": 0,
      "outline": "none",
      "padding": 0,
      "position": "absolute",
      "resize": "none",
      "translate": "0",
      "zIndex": 1,
    }
  }
  value="|--------------------------------------|
|--------------------------------------|
|--------------------------------------|
|--------------------------------------|
|--------------------------------------|
|--------------------------------------|"
/>
`;
